### DISCUSSION TITLE
New release: v0.22.0

### DISCUSSION THREAD

**Discussion by ObserverOfTime** [tree-sitter/py-tree-sitter#236] [Announcements]
**This release removes the old `Language.build_library` API.**

- [README](https://github.com/tree-sitter/py-tree-sitter/blob/v0.22.0/README.md)
- [Release notes](https://github.com/tree-sitter/py-tree-sitter/releases/tag/v0.22.0)
- [API documentation](https://tree-sitter.github.io/py-tree-sitter/)

**Comment by leotrs** [comment]
@ObserverOfTime This release introduced two braking changes:  `Language.build_library` is gone and the constructor `Language()` only takes a single integer argument (whereas before it took two strings, a path and a language name).

It is clear that the intention is to allow the user to load a pre-packaged and published language grammar. However, this has broken the workflow when developing an in-development not-yet-released (or private) language grammar.

Before v0.22.0, I would write my grammar, and use `tree-sitter generate` and `tree-sitter build` to get a `*.so` file. Then I would place this file in my `PATH` and simply use `Language(<path_to_so_file>, "my_lang_name")`. 

How can I do this now?

**Reply by ObserverOfTime** [reply]
We're already at 0.24 now and I've explained this many times in multiple issues and discussions.

**Reply by leotrs** [reply]
The only relevant discussion I can find is [this one](https://github.com/tree-sitter/py-tree-sitter/discussions/241#discussioncomment-9510930). If I understand correctly, I must build a wheel file from my tree-sitter-lang repository. Is this correct? How then do I pass this into the Language() constructor? I have not found any other documentation for this.

**Reply by ObserverOfTime** [reply]
https://github.com/tree-sitter/py-tree-sitter#install-languages

**Reply by leotrs** [reply]
Thanks @ObserverOfTime.

My 2 cents: in [the documentation](https://tree-sitter.github.io/tree-sitter/creating-parsers/4-external-scanners.html), the following paragraph could/should be amended to include that the file `tree-sitter-mylang/setup.py` _must also_ be modified to include a reference to `src/scanner.c`, if an external scanner exists.

> Then, add another C source file to your project. Its path must be src/scanner.c for the CLI to recognize it. Be sure to add this file to the sources section of your binding.gyp file so that it will be included when your project is compiled by Node.js and uncomment the appropriate block in your bindings/rust/build.rs file so that it will be included in your Rust crate.
